include/swift/Option/Options.td

```diff
@@ -399,15 +408,15 @@ def remark_skip_explicit_interface_build : Flag<["-"], "Rskip-explicit-interface
 
 def emit_tbd : Flag<["-"], "emit-tbd">,
   HelpText<"Emit a TBD file">,
-  Flags<[FrontendOption, NoInteractiveOption, SupplementaryOutput]>;
+  Flags<[FrontendOption, NoInteractiveOption, SupplementaryOutput, CacheInvariant]>;
```

Comment:
Do we always emit a TBD with caching, or is this because the TBD is in a separate job that is cached independently?  Same question applies to all the other `emit-X` options you have marked `CacheInvariant`.

Reply:
The comments for the declaration of `CacheInvariant` says:
```
// The option that is cache invariant. Change this option doesn't affect the
 // content of any output file. The candidates are options like: `-o` which only
 // affects the location of the output file, or `-emit-module` which defines
 // which action is performed.
```
The idea is the option only describes if an output exists or not, but doesn't affect other outputs, thus doesn't affect cache key for other outputs.